src/modules/home/components/LoggedInHome/LoggedInHome.tsx

```diff
@@ -128,59 +160,94 @@ const LoggedInHome: FC = () => {
+
+					{(active?.items || []).length > 0 && (
 						<section
 							className={clsx(
 								styles['c-hero__section'],
 								styles['c-hero__section--access']
 							)}
 						>
 							<div className={styles['c-hero__access-cards']}>
-								{accessGranted.map((room, i) => (
+								{(active?.items || []).map((visit, i) => (
 									<ReadingRoomCard
 										key={`hero-access-${i}`}
-										access={AccessGranted}
-										room={room}
+										access={{
+											granted: true,
+											until: asDate(visit.endAt),
+											from: asDate(visit.startAt),
+										}}
+										room={{
+											image: visit.spaceImage,
+											logo: visit.spaceLogo,
+											color: visit.spaceColor,
+											name: visit.spaceName,
+											info: visit.spaceInfo,
+										}}
 										type={ReadingRoomCardType.access}
 									/>
 								))}
 							</div>
 						</section>
 					)}
-					{planned.length > 0 && (
+
+					{(future?.items || []).length > 0 && (
 						<section className={clsx(styles['c-hero__section'])}>
 							<h5 className={clsx(styles['c-hero__section-title'], 'u-mb-16')}>
 								{t('modules/shared/components/hero/hero___geplande-bezoeken')}
 							</h5>
 							<div className={styles['c-hero__requests']}>
-								{planned.map((room, i) => (
+								{(future?.items || []).map((visit, i) => (
 									<ReadingRoomCard
 										key={`hero-planned-${i}`}
-										access={AccessGranted}
-										room={room}
+										access={{
+											granted: true,
+											until: asDate(visit.endAt),
+											from: asDate(visit.startAt),
+										}}
+										room={{
+											image: visit.spaceImage,
+											logo: visit.spaceLogo,
+											color: visit.spaceColor,
+											name: visit.spaceName,
+											info: visit.spaceInfo,
+										}}
 										type={ReadingRoomCardType.futureApproved}
 									/>
 								))}
 							</div>
 						</section>
 					)}
-					{requested.length > 0 && (
+
+					{(pending?.items || []).length > 0 && (
 						<section className={clsx(styles['c-hero__section'])}>
 							<h5 className={clsx(styles['c-hero__section-title'], 'u-mb-16')}>
 								{t('modules/shared/components/hero/hero___aanvragen')}
 							</h5>
 							<div className={styles['c-hero__requests']}>
-								{requested.map((room, i) => (
+								{(pending?.items || []).map((visit, i) => (
 									<ReadingRoomCard
 										key={`hero-requested-${i}`}
-										access={AccessGranted}
-										room={room}
+										access={{
+											granted: false,
+											pending: true,
+											until: asDate(visit.endAt),
+											from: asDate(visit.startAt),
+										}}
+										room={{
+											image: visit.spaceImage,
+											logo: visit.spaceLogo,
+											color: visit.spaceColor,
+											name: visit.spaceName,
+											info: visit.spaceInfo,
+										}}
 										type={ReadingRoomCardType.futureRequested}
 									/>
```

Comment:
extract common logic:
```
{renderVisitRequestCard(visit, ReadingRoomCardType.futureRequested)}
```

And reuse it 3 times

Reply:
Fixed in https://github.com/viaacode/hetarchief-client/pull/231/commits/020a7c962b81a01d16a2828bd7307c63ecab3828